packages/workspace/src/common/workspace-protocol.ts

```diff
@@ -32,4 +32,41 @@ export interface WorkspaceServer {
      */
     setRoot(uri: string): Promise<void>;
 
+    /**
```

Comment:
discussed with @svenefftinge: we should rename `getRoot/setRoot` to `getWorkspace/setWorkspace`. It returns a path to a workspace folder or a file. In the case of a folder `WorkspaceService` should work as before. In the case of a file `WorkspaceService` should watch, read and modify the file from the frontend via `FileSystem` and `FileSystemWatcher`.

Reply:
That distinguishes the `workspace` from `root`. 
Changes are made. @akosyakov